fvm/state/accounts.go

```diff
@@ -82,6 +82,24 @@ func (a *StatefulAccounts) AllocateStorageIndex(address flow.Address) (atree.Sto
 	copy(index[:], indexBytes[:8])
 	newIndexBytes := index.Next()
 
+	// store nil so that the setValue for new allocated slabs would be faster
+	// and won't do ledger getValue for every new slabs (currently happening to compute storage size changes)
+	// this way the getValue would load this value from deltas
+	// TODO: update this code to use atree.SlabIndexToLedgerKey method when the new version of atree is used
+	key := atree.LedgerBaseStorageSlabPrefix + string(index[:])
+
+	err = a.stateHolder.State().Set(string(address.Bytes()), "", key, []byte{}, false)
+	if err != nil {
+		return atree.StorageIndex{}, fmt.Errorf("failed to store empty value for newly allocated storage index: %w", err)
+	}
+	// update the storage size considering new keys has been allocated with empty value
+	oldSize, err := a.GetStorageUsed(address)
+	if err != nil {
+		return atree.StorageIndex{}, fmt.Errorf("failed to store empty value for newly allocated storage index: %w", err)
+	}
+	a.setStorageUsed(address, oldSize+uint64(RegisterSize(address, false, key, []byte{})))
```

Comment:
Can the two byte slice allocations, `[]byte{}`, be just replaced with `nil`?

Reply:
I just noticed this won't work since RegisterSize would ignore the size if the value size is empty, which is not really great cc: @janezpodhostnik

Reply:
so maybe we don't even need to set the storage used for nil values :-?